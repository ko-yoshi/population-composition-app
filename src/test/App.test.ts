import { shallowMount } from '@vue/test-utils'
import { expect, test, vi, describe } from 'vitest'
import App from '~/app.vue'

vi.mock('#app', () => ({}))

describe('App.vue', () => {
  const wrapper = shallowMount(App)

  test('ヘッダー確認', () => {
    const header = wrapper.find('.header')
    expect(header.exists()).toBe(true)
    expect(header.find('h1').text()).toBe('都道府県別人口推移グラフ')
  })

  test('PrefecturesAreaのレンダリング確認', () => {
    expect(wrapper.findComponent({ name: 'PrefecturesArea' }).exists()).toBe(true)
  })

  test('PopulationTypeAreaのレンダリング確認', () => {
    expect(wrapper.findComponent({ name: 'PopulationTypeArea' }).exists()).toBe(true)
  })

  test('LineChartのレンダリング確認', () => {
    expect(wrapper.findComponent({ name: 'LineChart' }).exists()).toBe(true)
  })

  test('snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
