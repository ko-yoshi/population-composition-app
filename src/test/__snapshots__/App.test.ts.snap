// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App.vue > snapshot 1`] = `
"<main data-v-37ddea6c=\\"\\" class=\\"main\\">
  <section data-v-37ddea6c=\\"\\" class=\\"header\\">
    <h1 data-v-37ddea6c=\\"\\">都道府県別人口推移グラフ</h1>
  </section>
  <section data-v-37ddea6c=\\"\\">
    <prefectures-area-stub data-v-37ddea6c=\\"\\"></prefectures-area-stub>
  </section>
  <section data-v-37ddea6c=\\"\\" class=\\"population-type-area\\">
    <population-type-area-stub data-v-37ddea6c=\\"\\"></population-type-area-stub>
  </section>
  <section data-v-37ddea6c=\\"\\" class=\\"line-chart-area\\">
    <line-chart-stub data-v-37ddea6c=\\"\\"></line-chart-stub>
  </section>
</main>"
`;
